pkgs/top-level/cuda-packages.nix

```diff
@@ -127,30 +143,53 @@ let
           directory = ../development/cuda-modules/packages;
         }
       )
-      (callPackage ../development/cuda-modules/cuda/extension.nix { inherit cudaVersion; })
-      (import ../development/cuda-modules/cuda/overrides.nix)
-      (callPackage ../development/cuda-modules/generic-builders/multiplex.nix {
-        inherit cudaVersion flags mkVersionedPackageName;
+      (import ../development/cuda-modules/cuda/extension.nix { inherit cudaVersion fixups lib; })
+      (import ../development/cuda-modules/generic-builders/multiplex.nix {
+        inherit
+          cudaVersion
+          flags
+          lib
+          mkVersionedPackageName
+          stdenv
+          ;
         pname = "cudnn";
+        redistName = "cudnn";
         releasesModule = ../development/cuda-modules/cudnn/releases.nix;
         shimsFn = ../development/cuda-modules/cudnn/shims.nix;
-        fixupFn = ../development/cuda-modules/cudnn/fixup.nix;
       })
-      (callPackage ../development/cuda-modules/cutensor/extension.nix {
-        inherit cudaVersion flags mkVersionedPackageName;
+      (import ../development/cuda-modules/cutensor/extension.nix {
+        inherit
+          cudaVersion
+          flags
+          lib
+          mkVersionedPackageName
+          stdenv
+          ;
       })
-      (callPackage ../development/cuda-modules/cusparselt/extension.nix {
-        inherit cudaVersion flags mkVersionedPackageName;
+      (import ../development/cuda-modules/cusparselt/extension.nix {
+        inherit
+          cudaVersion
+          flags
+          lib
+          mkVersionedPackageName
+          stdenv
+          ;
       })
-      (callPackage ../development/cuda-modules/generic-builders/multiplex.nix {
-        inherit cudaVersion flags mkVersionedPackageName;
+      (import ../development/cuda-modules/generic-builders/multiplex.nix {
+        inherit
+          cudaVersion
+          flags
+          lib
+          mkVersionedPackageName
+          stdenv
+          ;
         pname = "tensorrt";
+        redistName = "tensorrt";
         releasesModule = ../development/cuda-modules/tensorrt/releases.nix;
         shimsFn = ../development/cuda-modules/tensorrt/shims.nix;
-        fixupFn = ../development/cuda-modules/tensorrt/fixup.nix;
       })
-      (callPackage ../development/cuda-modules/cuda-samples/extension.nix { inherit cudaVersion; })
-      (callPackage ../development/cuda-modules/cuda-library-samples/extension.nix { })
+      (import ../development/cuda-modules/cuda-samples/extension.nix { inherit cudaVersion lib stdenv; })
+      (import ../development/cuda-modules/cuda-library-samples/extension.nix { inherit lib stdenv; })
```

Comment:
Why these?

Reply:
Why convert them to `import`? Because we benefit directly from import caching, rather than hiding `import` inside of a `callPackage`.

Reply:
Ah so the difference is it's easier to prove the result doesn't depend on `self`? I wonder if there's any source other than C++ sources that actually clearly lays out our memoization rules...

Reply:
Nix maintains a cache for parsed files; see https://github.com/NixOS/nix/blob/1e822bd4149a8bce1da81ee2ad9404986b07914c/src/libexpr/primops.cc#L282 and https://github.com/NixOS/nix/blob/1e822bd4149a8bce1da81ee2ad9404986b07914c/src/libexpr/eval.cc#L1122

Reply:
But that doesn't necessarily mean there's a difference with `callPackage`

Reply:
Only that we can lift it up to the top, drop `callPackage`'s (small) overhead, and don't need to worry about them being `override`-able.

Reply:
> can lift it up to the top

This, I didn't know how to phrase it, but this is what I meant by "prove no dependency on `self`"